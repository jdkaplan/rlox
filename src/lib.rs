macro_rules! debug_log_gc {
    ($($arg:tt)*) => {{
        #[cfg(feature = "log_gc")]
        println!($($arg)*);
    }};
}

mod chunk;
mod compiler;
mod gc;
mod heap;
mod object;
mod scanner;
mod table;
mod value;
mod vm;

pub use vm::{InterpretError, InterpretResult, Vm};

pub const U8_COUNT: usize = (u8::MAX as usize) + 1;
pub const FRAMES_MAX: usize = 64;
pub const STACK_MAX: usize = FRAMES_MAX * U8_COUNT;
