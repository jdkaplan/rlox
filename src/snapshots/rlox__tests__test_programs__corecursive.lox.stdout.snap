---
source: src/lib.rs
description: "fun even(n) {\n    if (n == 0) return true;\n    return odd(n - 1);\n}\n\nfun odd(n) {\n    if (n == 0) return false;\n    return even(n - 1);\n}\n\nfor (var i = 0; i < 10; i = i + 1) {\n    print even(i);\n}\n"
input_file: test_programs/corecursive.lox
---
true
false
true
false
true
false
true
false
true
false

